src/astro/files/__init__.py

```diff
@@ -14,3 +16,26 @@ def check_if_connection_exists(conn_id: str) -> bool:
     """
     BaseHook.get_connection(conn_id)
     return True
+
+
+def get_file_list(path: str, conn_id: str, **kwargs) -> TaskDecorator:
+    """
+    List the file path from the filesystem storage based on given path pattern
+
+    Supported filesystem: Local, HTTP, S3, GCS
+
+    :param conn_id: Airflow connection id
+    :param path: Path pattern to the file in the filesystem/Object stores
+    """
+    if "task_id" in kwargs:
+        task_id = kwargs["task_id"]
+        del kwargs["task_id"]
+    else:
+        task_id = get_unique_task_id(task_id="get_file_list")
```

Comment:
We don't need this method of airflow handles this internally. Will append number if there are duplicate task names.

Reply:
No, if you place your task with a duplicate id in a DAG airflow throw exception and do not resolve it.